artiq/coredevice/comm_kernel.py

```diff
@@ -686,6 +687,14 @@ def read_exception_string():
         else:
             python_exn_type = embedding_map.retrieve_object(core_exn.id)
 
+        if any(python_exn_type == exn_type
+                for exn_type in [exceptions.RTIOUnderflow,
+                                 exceptions.RTIOOverflow,
+                                 exceptions.RTIODestinationUnreachable]):
+            channel_involved = self.get_channel_name(nested_exceptions[0][2][0])
+            nested_exceptions[-1][1] = nested_exceptions[-1][1].replace(
```

Comment:
As I already told you multiple times, stop mutating data when not necessary. Why not just create a new variable there? It would also avoid the clumsy repetition of ``nested_exceptions[-1][1]``.